tests/decomposition/test_symbolic_decomposition.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests the decomposition rules defined for symbolic operations other than controlled."""
+
+import pytest
+
+import pennylane as qml
+from pennylane.decomposition.resources import Resources
+from pennylane.decomposition.symbolic_decomposition import (
+    AdjointDecomp,
+    adjoint_adjoint_decomp,
+    adjoint_controlled_decomp,
+    has_adjoint_decomp,
+)
+from tests.decomposition.conftest import to_resources
+
+
+class TestAdjointDecompositionRules:
+    """Tests the decomposition rules defined for the adjoint of operations."""
+
+    def test_adjoint_adjoint(self):
+        """Tests that the adjoint of an adjoint cancels out."""
+
+        op = qml.adjoint(qml.adjoint(qml.RX(0.5, wires=0)))
+
+        with qml.queuing.AnnotatedQueue() as q:
+            adjoint_adjoint_decomp(*op.parameters, wires=op.wires, **op.hyperparameters)
+
+        assert q.queue == [qml.RX(0.5, wires=0)]
+        assert adjoint_adjoint_decomp.compute_resources(**op.resource_params) == to_resources(
+            {qml.RX: 1}
+        )
+
+    @pytest.mark.jax
+    def test_adjoint_adjoint_capture(self):
+        """Tests that the adjoint of an adjoint works with capture."""
+
+        from pennylane.tape.plxpr_conversion import CollectOpsandMeas
+
+        op = qml.adjoint(qml.adjoint(qml.RX(0.5, wires=0)))
+
+        capture_enabled = qml.capture.enabled()
+        qml.capture.enable()
+
+        def circuit():
+            adjoint_adjoint_decomp(*op.parameters, wires=op.wires, **op.hyperparameters)
+
+        plxpr = qml.capture.make_plxpr(circuit)()
+        collector = CollectOpsandMeas()
+        collector.eval(plxpr.jaxpr, plxpr.consts)
+        assert collector.state["ops"] == [qml.RX(0.5, wires=0)]
+
+        if not capture_enabled:
+            qml.capture.disable()
```

Comment:
This test surprised me. Should there be more capture-related tests in this line of PRs? :thinking: What makes `adjoint` stand out to require this test?

Reply:
The decomposition of `op = qml.adjoint(qml.T)` is `op.base`, but since the decomposition is a qfunc, if we write
```
def decomp(...):
    qml.apply(base)
```
it doesn't get captured by program capture. The solution ended up being
```
def decomp(...):
    base._unflatten(*base._flatten())
```
which both queues the op in a queueing context, and gets captured properly. This test is to make sure of that.

Reply:
Ah, I noticed the function having this form but forgot that it does so because of capture :sweat_smile: Thanks! :+1: